File: 4_bresenham_line.py
import matplotlib.pyplot as plt

x1, y1 = 2, 3
x2, y2 = 10, 8

dx = abs(x2 - x1)
dy = abs(y2 - y1)

p = 2*dy - dx
x = x1
y = y1

x_points = []
y_points = []

while x <= x2:
    x_points.append(x)
    y_points.append(y)
    x += 1
    if p < 0:
        p += 2*dy
    else:
        y += 1
        p += 2*(dy - dx)

plt.plot(x_points, y_points, marker='o')
plt.grid()
plt.title("Bresenham Line Algorithm")
plt.show()

File: 4_bresenham_line.m
clc; clear; close all;

x1=2; y1=3;
x2=10; y2=8;

dx = abs(x2-x1);
dy = abs(y2-y1);

p = 2*dy - dx;
x = x1;
y = y1;

while x <= x2
    plot(x,y,'o'); hold on;
    x = x + 1;
    if p < 0
        p = p + 2*dy;
    else
        y = y + 1;
        p = p + 2*(dy-dx);
    end
end

grid on;
title('Bresenham Line Algorithm');
